ash-window/src/lib.rs

```diff
@@ -122,10 +123,16 @@ pub unsafe fn create_surface(
 /// The returned extensions will include all extension dependencies.
 pub fn enumerate_required_extensions(
     window_handle: &dyn HasRawWindowHandle,
-) -> VkResult<Vec<&'static CStr>> {
+) -> VkResult<&'static [*const c_char]> {
     let extensions = match window_handle.raw_window_handle() {
         #[cfg(target_os = "windows")]
-        RawWindowHandle::Windows(_) => vec![khr::Surface::name(), khr::Win32Surface::name()],
+        RawWindowHandle::Windows(_) => {
+            const WINDOWS_EXTS: [*const c_char; 2] = [
+                khr::Surface::name().as_ptr(),
+                khr::Win32Surface::name().as_ptr(),
+            ];
+            &WINDOWS_EXTS
```

Comment:
I don't think it's actually necessary to have a separate `const` declaration, so we could avoid that boilerplate here and below. `'static` promotion applies regardless.

Reply:
Is there any chance you could elaborate on what you have in mind? I've been surprised in the past that Rust was smart enough to promote `&()` into a `&'static ()` but I had trouble getting that to work here. I tried this but Rust interprets it as a temporary value:

```rust
RawWindowHandle::Windows(_) => &[
    khr::Surface::name().as_ptr(),
    khr::Win32Surface::name().as_ptr(),
],
```

Is there another approach I should take?